prusti-tests/tests/verify_overflow/pass/competitve_programming/dp/TheJackpot_10684_failing.rs

```diff
@@ -0,0 +1,223 @@
+//https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&category=24&page=show_problem&problem=1625
+use prusti_contracts::*;
+
+pub struct VecWrapperI32 {
+    v: Vec<i32>,
+}
+
+impl VecWrapperI32 {
+    #[trusted]
+    #[pure]
+    #[ensures (0 <= result)]
+    pub fn len(&self) -> usize {
+        self.v.len()
+    }
+
+    #[trusted]
+    #[ensures (result.len() == 0)]
+    pub fn new() -> Self {
+        VecWrapperI32 { v: Vec::new() }
+    }
+
+    #[trusted]
+    #[pure]
+    #[requires (0 <= index && index < self.len())]
+    pub fn lookup(&self, index: usize) -> i32 {
+        self.v[index]
+    }
+
+    #[trusted]
+    #[ensures (self.len() == old(self.len()) + 1)]
+    #[ensures (self.lookup(old(self.len())) == value)]
+    #[ensures (forall(|i: usize| (0 <= i && i < old(self.len())) ==>
+                    self.lookup(i) == old(self.lookup(i))))]
+    pub fn push(&mut self, value: i32) {
+        self.v.push(value);
+    }
+}
```

Comment:
Does this work with an `#[extern_spec]` implementation, like [here](https://github.com/viperproject/prusti-dev/blob/21d7d5616b581fdf9741bd1da44f42dd1f75321d/prusti-tests/tests/verify/pass/extern-spec/vec-3.rs#L21-L42)?

Reply:
Does `#[extern_spec]` allow adding a pure `lookup` function? (As far as I remember there were some serious problems with supporting the `Index` and `IndexMut` traits.)

Reply:
I tried but I also couldn't find a way to support lookup.

Reply:
Ah yes, you're right, the encoding panics when trying to call such a function. I suppose you could write a trusted wrapper only for the lookup function but then it's more messy.